pandas/core/frame.py

```diff
@@ -816,14 +816,14 @@ def itertuples(self, index=True, name="Pandas"):
         Pandas(Index='b', col1=2, col2=0.20000000000000001)
 
         """
-        arrays = []
+        iterators = []
         fields = []
         if index:
-            arrays.append(self.index)
+            iterators.append(self.index)
             fields.append("Index")
 
         # use integer indexing because of possible duplicate column names
-        arrays.extend(self.iloc[:, k] for k in range(len(self.columns)))
+        iterators.extend(self.iloc[:, k] for k in range(len(self.columns)))
```

Comment:
I believe `iterators` can also be evaluated lazily by using `itertools.chain`. Something along the lines of:

```
iterators = itertools.chain([self.index], (self.iloc[:, k] for k in range(len(self.columns)))
```

Reply:
Yea, but I do not think this is necessary, because those are per-column and have to be done immediately afterwards even for the first row. So you move forcing execution from one line in a function to another line in the same function. So later on we do `zip(*iterators)` which forces `iterators` to be made into a list. If we would manually zip things, then we could go without that. But passing `*iterators` forces the list. It is also better that any exception is thrown there and not later inside that other `try/catch`.

So I do not think this is necessary.